src/components/cl/hier/alltoall/alltoall.c

```diff
@@ -45,42 +47,43 @@ UCC_CL_HIER_PROFILE_FUNC(ucc_status_t, ucc_cl_hier_alltoall_init,
         args.args.mask  = UCC_COLL_ARGS_FIELD_FLAGS;
         args.args.flags = 0;
     }
-    args.args.flags |= UCC_COLL_ARGS_FLAG_CONTIG_SRC_BUFFER |
-                       UCC_COLL_ARGS_FLAG_CONTIG_DST_BUFFER;
-    team_size = UCC_CL_TEAM_SIZE(cl_team);
 
-    status =
-        ucc_mc_alloc(&h, sizeof(int) * team_size * 2, UCC_MEMORY_TYPE_HOST);
+    args.args.flags |= UCC_COLL_ARGS_FLAG_CONTIG_SRC_BUFFER |
+                       UCC_COLL_ARGS_FLAG_CONTIG_DST_BUFFER |
+                       UCC_COLL_ARGS_FLAG_COUNT_64BIT |
+                       UCC_COLL_ARGS_FLAG_DISPLACEMENTS_64BIT;
+    status = ucc_mc_alloc(&h, sizeof(uint64_t) * tsize * 2,
+                          UCC_MEMORY_TYPE_HOST);
     if (ucc_unlikely(UCC_OK != status)) {
         cl_error(team->context->lib,
                  "failed to allocate %zd bytes for full counts",
-                 sizeof(int) * team_size * 2);
+                 sizeof(uint64_t) * tsize * 2);
```

Comment:
theoretically it should be `ucc_count_t * tsize + ucc_aint_t * tsize`, right ?

Reply:
yes, but in practice we never use 32b count and and 32b offset. I just didn't want to make code complicated by making separate flows for every combination of count and offset size. uint64_t should work always however it adds small memory overhead.

Reply:
ok no problem, that makes sense, thanks for explaining